src/NewContactDialog.jsx

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+var React = require('react');
+var ReactDOM = require('react-dom');
+var ReactBootstrap = require('react-bootstrap');
+var Modal = ReactBootstrap.Modal;
+var Button = ReactBootstrap.Button;
+var FormControl = ReactBootstrap.FormControl;
+var FormGroup = ReactBootstrap.FormGroup;
+var ControlLabel = ReactBootstrap.ControlLabel;
+var $ = require('jquery');
+
+var common = require('./common.jsx');
+
+module.exports = React.createClass({
+    getInitialState() {
+        return {
+            name: '',
+            email: '',
+            type: 'people',
+            message: '',
+            msg_type: ''
+        };
+    },
+
+    setType(type) {
+        this.setState({type: type});
+    },
+
+    setError(msg) {
+        this.setState({message: msg, msg_type: 'Error'});
+    },
+
+    submit() {
+        if (!this.state.name || !this.state.email) {
+            this.setState({error: 'Both fields are required'});
```

Comment:
No error is visible when either field is empty. Did you mean `this.setError(...)`?